[Automation] javac compile fails for org.jetbrains.exposed:exposed-core:1.3.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25860675378/job/75990475408
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
t.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.


* What went wrong:
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 17s

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 32s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.3.0 at 2026-05-14T12:52:58Z with exit code 1 after 32s
❌ ERROR: 'native-image run' for 1.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.3.0 at 2026-05-14T12:52:58Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 944ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :jar

> Task :compileTestKotlin FAILED
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date

BUILD FAILED in 2s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.3.0 at 2026-05-14T12:53:03Z with exit code 1 after 5s
❌ ERROR: 'javac compile' for 1.3.0 failed with exit code 1.
FAILED[javac compile][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25860675378/job/75990475428
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


BUILD FAILED in 14s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 29s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.3.0 at 2026-05-14T12:52:41Z with exit code 1 after 30s
❌ ERROR: 'native-image run' for 1.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.3.0 at 2026-05-14T12:52:41Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 913ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar

> Task :compileTestKotlin FAILED
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.3.0 at 2026-05-14T12:52:46Z with exit code 1 after 5s
❌ ERROR: 'javac compile' for 1.3.0 failed with exit code 1.
FAILED[javac compile][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25860675378/job/75990475419
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
t:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 19s
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jetbrains.exposed:exposed-core:1.0.0-beta-4 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 33s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.3.0 at 2026-05-14T12:53:12Z with exit code 1 after 33s
❌ ERROR: 'native-image run' for 1.3.0 failed with exit code 1.
SOFT-FAIL[native-image run][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.3.0" ./gradlew clean test -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
========================================================================================
 org.jetbrains.exposed:exposed-core:1.0.0-beta-4:1.3.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.3.0 at 2026-05-14T12:53:12Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 739ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
> Task :compileKotlin NO-SOURCE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :jar

> Task :compileTestKotlin FAILED
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:19:8 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:146:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:28 'fun <T : Comparable<T>, S : T?> Expression<in S>.less(other: Expression<in S>): LessOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:147:77 'fun <T : String?> Expression<T>.like(pattern: String): LikeEscapeOp' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
11 actionable tasks: 2 executed, 9 up-to-date
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:151:65 'fun <T> ExpressionWithColumnType<T>.inList(list: Iterable<T>): InListOrNotInListBaseOp<T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:31 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:71 'fun <T> ExpressionWithColumnType<T>.plus(t: T): PlusOp<T, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:76 'fun <T, S : T> ExpressionWithColumnType<T>.times(other: Expression<S>): TimesOp<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:154:93 'fun <T : Number?, S : T> ExpressionWithColumnType<T>.rem(t: S): ModOp<T, S, T>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:157:35 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:158:13 'fun case(): Case' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:159:37 'fun <T> ExpressionWithColumnType<T>.eq(t: T): Op<Boolean>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:164:30 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:165:13 'fun <T, S : T?> coalesce(expr: ExpressionWithColumnType<S>, alternate: Expression<out T>, vararg others: Expression<out T>): Coalesce<T, S>' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:173:27 'object SqlExpressionBuilder : ISqlExpressionBuilder' is deprecated. This builder object will continue to be phased out following release 1.0.0. All expression builder methods previously restricted to this object have also been deprecated in favor of equivalent top-level functions, making this object useless as a receiver in any scope. It will no longer be necessary to import each individual method when used outside a scoped block, and on demand imports will now be possible via 'import org.jetbrains.exposed.v1.core.*', if required.
e: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jetbrains.exposed/exposed-core/1.0.0-beta-4/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt:174:13 'fun rowNumber(): RowNumber' is deprecated. This interface method will be removed following release 1.0.0 and should be replaced with the equivalent top-level function.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.btapi.BuildToolsApiCompilationWork
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.3.0 at 2026-05-14T12:53:16Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 1.3.0 failed with exit code 1.
FAILED[javac compile][1.3.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.3.0" ./gradlew clean compileTestJava -Pcoordinates="org.jetbrains.exposed:exposed-core:1.0.0-beta-4"]
```

--- a/metadata/org.jetbrains.exposed/exposed-core/1.3.0/reachability-metadata.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/1.3.0/reachability-metadata.json
@@ -1,0 +1,376 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.SqlServerVectorBinder"
+      },
+      "type": "com.microsoft.sqlserver.jdbc.SQLServerResource"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "java.lang.Class[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "java.lang.annotation.Annotation[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "java.lang.annotation.Annotation[][]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "java.lang.reflect.Field[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "java.util.logging.ConsoleHandler"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "java.util.logging.Handler[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "java.util.logging.SimpleFormatter"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "javax.management.MBeanAttributeInfo[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "javax.management.MBeanOperationInfo"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "javax.management.MBeanOperationInfo[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "javax.management.openmbean.OpenMBeanOperationInfoSupport"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.Any"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.Boolean"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.Int"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.SafePublicationLazyImpl"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.String"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.coroutines.Continuation[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.ReflectionFactoryImpl"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.descriptors.TypeParameterDescriptor[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.km.jvm.internal.JvmMetadataExtensions"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.load.java.ErasedOverridabilityCondition"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.load.java.FieldOverridabilityCondition"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.load.java.JavaIncompatibilityRulesOverridabilityCondition"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.resolve.scopes.MemberScope[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "kotlin.reflect.jvm.internal.impl.types.TypeProjection[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.DatabaseConfig$Builder"
+      },
+      "type": "kotlinx.coroutines.EventLoopImplBase"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.DatabaseConfig$Builder"
+      },
+      "type": "kotlinx.coroutines.internal.LimitedDispatcher"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.DatabaseConfig$Builder"
+      },
+      "type": "kotlinx.coroutines.internal.LockFreeTaskQueue"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.DatabaseConfig$Builder"
+      },
+      "type": "kotlinx.coroutines.internal.LockFreeTaskQueueCore"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.DatabaseConfig$Builder"
+      },
+      "type": "kotlinx.coroutines.scheduling.CoroutineScheduler"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.SqlServerVectorBinder"
+      },
+      "type": "microsoft.sql.Vector",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": [
+            "int",
+            "int",
+            "java.lang.Object[]"
+          ]
+        },
+        {
+          "name": "getData",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "oracle.jdbc.diagnostics.DiagnosticsMXBean"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "oracle.jdbc.diagnostics.OracleDiagnosticsMXBean"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "oracle.jdbc.driver.SQLStateMapping[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": "oracle.sql.VECTOR",
+      "methods": [
+        {
+          "name": "ofFloat32Values",
+          "parameterTypes": [
+            "java.lang.Object"
+          ]
+        },
+        {
+          "name": "ofFloat64Values",
+          "parameterTypes": [
+            "java.lang.Object"
+          ]
+        },
+        {
+          "name": "ofInt8Values",
+          "parameterTypes": [
+            "java.lang.Object"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.ColumnType",
+      "methods": [
+        {
+          "name": "getNullable",
+          "parameterTypes": []
+        },
+        {
+          "name": "setNullable",
+          "parameterTypes": [
+            "boolean"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table$PrimaryKey"
+      },
+      "type": "org.jetbrains.exposed.v1.core.Column[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.FunctionBuilderKt"
+      },
+      "type": "org.jetbrains.exposed.v1.core.Expression[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.IColumnType"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.IntegerColumnType",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.statements.api.PreparedStatementApi"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": "org.jetbrains.exposed.v1.core.statements.api.RowApi"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Transaction"
+      },
+      "type": "sun.security.provider.NativePRNG"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Transaction"
+      },
+      "type": "sun.security.provider.SHA"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": {
+        "proxy": [
+          "java.lang.annotation.Retention"
+        ]
+      }
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.OracleVectorBinder"
+      },
+      "type": {
+        "proxy": [
+          "javax.management.MXBean"
+        ]
+      }
+    },
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "type": {
+        "proxy": [
+          "kotlin.Metadata"
+        ]
+      }
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.jetbrains.exposed.v1.core.Table"
+      },
+      "glob": "META-INF/services/kotlin.reflect.jvm.internal.impl.util.ModuleVisibilityHelper"
+    }
+  ]
+}

--- a/metadata/org.jetbrains.exposed/exposed-core/index.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/index.json
@@ -19,6 +19,19 @@
     ]
   },
   {
+    "metadata-version" : "1.3.0",
+    "language" : {
+      "name" : "kotlin",
+      "version" : "2.0"
+    },
+    "tested-versions" : [
+      "1.3.0"
+    ],
+    "allowed-packages" : [
+      "org.jetbrains.exposed.v1"
+    ]
+  },
+  {
     "metadata-version" : "1.2.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/jetbrains/exposed/exposed-core/$version$/exposed-core-$version$-sources.jar",
     "repository-url" : "https://github.com/JetBrains/Exposed",

--- a/metadata/org.jetbrains.exposed/exposed-core/index.json
+++ b/metadata/org.jetbrains.exposed/exposed-core/index.json
@@ -20,6 +20,11 @@
   },
   {
     "metadata-version" : "1.3.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/jetbrains/exposed/exposed-core/$version$/exposed-core-$version$-sources.jar",
+    "repository-url" : "https://github.com/JetBrains/Exposed",
+    "test-code-url" : "https://github.com/JetBrains/Exposed/tree/$version$/exposed-tests/src/test",
+    "documentation-url" : "https://github.com/JetBrains/Exposed/blob/$version$/docs/api/exposed-core/index.html",
+    "description" : "Exposed Core is the base module of JetBrains Exposed, a Kotlin SQL/ORM framework for building type-safe database code. It provides the SQL DSL primitives, schema and table model, expressions, statements, and dialect abstractions used by Exposed's JDBC, R2DBC, and DAO modules.",
     "language" : {
       "name" : "kotlin",
       "version" : "2.0"

--- a/stats/org.jetbrains.exposed/exposed-core/1.3.0/execution-metrics.json
+++ b/stats/org.jetbrains.exposed/exposed-core/1.3.0/execution-metrics.json
@@ -1,0 +1,93 @@
+{
+  "fix_javac_fail:2026-05-19": {
+    "timestamp": "2026-05-19T15:03:09.453475Z",
+    "library": "org.jetbrains.exposed:exposed-core:1.3.0",
+    "starting_commit": "f8f3a5fed43ba2479bb9b5ac8c0eb68642a96dca",
+    "ending_commit": "4df79327ed60813667d50eef87668d67186b543b",
+    "previous_library": "org.jetbrains.exposed:exposed-core:1.0.0-beta-4",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.3.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 11,
+            "totalCalls": 11
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 11,
+        "totalCalls": 11
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 5200,
+          "missed": 56445,
+          "ratio": 0.084354,
+          "total": 61645
+        },
+        "line": {
+          "covered": 831,
+          "missed": 7571,
+          "ratio": 0.098905,
+          "total": 8402
+        },
+        "method": {
+          "covered": 381,
+          "missed": 3176,
+          "ratio": 0.107113,
+          "total": 3557
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.0.0-beta-4",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4466,
+          "missed": 50240,
+          "ratio": 0.081636,
+          "total": 54706
+        },
+        "line": {
+          "covered": 710,
+          "missed": 6615,
+          "ratio": 0.096928,
+          "total": 7325
+        },
+        "method": {
+          "covered": 339,
+          "missed": 2727,
+          "ratio": 0.110568,
+          "total": 3066
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 195060,
+      "cached_input_tokens_used": 3131392,
+      "output_tokens_used": 19104,
+      "iterations": 3,
+      "cost_usd": 2.1388,
+      "tested_library_loc": 831,
+      "metadata_entries": 59,
+      "previous_library_metadata_entries": 5,
+      "code_coverage_percent": 9.89,
+      "previous_library_coverage_percent": 9.69
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/OracleVectorBinderTest.kt",
+      "metadata_file": "metadata/org.jetbrains.exposed/exposed-core/1.3.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.jetbrains.exposed/exposed-core/1.3.0/stats.json
+++ b/stats/org.jetbrains.exposed/exposed-core/1.3.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "1.3.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 11,
+          "totalCalls" : 11
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 11,
+      "totalCalls" : 11
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5200,
+        "missed" : 56445,
+        "ratio" : 0.084354,
+        "total" : 61645
+      },
+      "line" : {
+        "covered" : 831,
+        "missed" : 7571,
+        "ratio" : 0.098905,
+        "total" : 8402
+      },
+      "method" : {
+        "covered" : 381,
+        "missed" : 3176,
+        "ratio" : 0.107113,
+        "total" : 3557
+      }
+    }
+  } ]
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/.gitignore
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
@@ -14,6 +14,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 dependencies {
     testImplementation "org.jetbrains.exposed:exposed-core:$libraryVersion"
     testImplementation 'com.oracle.database.jdbc:ojdbc11:23.7.0.25.01'
+    testImplementation 'com.microsoft.sqlserver:mssql-jdbc:13.2.1.jre11'
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "org.jetbrains.kotlin.jvm" version "2.3.20"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.jetbrains.exposed:exposed-core:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+kotlin {
+    jvmToolchain(21)
+}
+
+compileTestKotlin {
+    kotlinOptions.jvmTarget = "21"
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/build.gradle
@@ -13,6 +13,7 @@ String libraryVersion = tck.testedLibraryVersion.get()
 
 dependencies {
     testImplementation "org.jetbrains.exposed:exposed-core:$libraryVersion"
+    testImplementation 'com.oracle.database.jdbc:ojdbc11:23.7.0.25.01'
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/gradle.properties
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.jetbrains.exposed:exposed-core:1.3.0
+library.version = 1.3.0
+metadata.dir = org.jetbrains.exposed/exposed-core/1.3.0/

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/settings.gradle
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.jetbrains.exposed.exposed-core_tests'

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
@@ -16,17 +16,26 @@ import org.jetbrains.exposed.v1.core.Index
 import org.jetbrains.exposed.v1.core.Join
 import org.jetbrains.exposed.v1.core.QueryBuilder
 import org.jetbrains.exposed.v1.core.Schema
-import org.jetbrains.exposed.v1.core.SqlExpressionBuilder
 import org.jetbrains.exposed.v1.core.Table
 import org.jetbrains.exposed.v1.core.Version
 import org.jetbrains.exposed.v1.core.WindowFrameBound
 import org.jetbrains.exposed.v1.core.alias
 import org.jetbrains.exposed.v1.core.and
 import org.jetbrains.exposed.v1.core.append
+import org.jetbrains.exposed.v1.core.case
+import org.jetbrains.exposed.v1.core.coalesce
+import org.jetbrains.exposed.v1.core.eq
+import org.jetbrains.exposed.v1.core.inList
 import org.jetbrains.exposed.v1.core.intLiteral
 import org.jetbrains.exposed.v1.core.intParam
+import org.jetbrains.exposed.v1.core.less
+import org.jetbrains.exposed.v1.core.like
+import org.jetbrains.exposed.v1.core.plus
+import org.jetbrains.exposed.v1.core.rem
+import org.jetbrains.exposed.v1.core.rowNumber
 import org.jetbrains.exposed.v1.core.stringLiteral
 import org.jetbrains.exposed.v1.core.stringParam
+import org.jetbrains.exposed.v1.core.times
 import org.jetbrains.exposed.v1.exceptions.DuplicateColumnException
 import org.junit.jupiter.api.Test
 
@@ -143,39 +152,31 @@ public class ExposedCoreTest {
 
     @Test
     fun expressionDslBuildsComparisonListsAliasesCasesAndWindows(): Unit {
-        val comparison = with(SqlExpressionBuilder) {
-            (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
-        }
+        val comparison = (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
         assertThat(comparison.toString()).isEqualTo("(1 < 3) AND ('kotlin' LIKE 'kot%')")
 
-        val inList = with(SqlExpressionBuilder) { intLiteral(2) inList listOf(3, 1, 2) }
+        val inList = intLiteral(2) inList listOf(3, 1, 2)
         assertThat(inList.toString()).isEqualTo("2 IN (3, 1, 2)")
 
-        val arithmetic = with(SqlExpressionBuilder) { ((intLiteral(5) + 4) * intLiteral(2)) % 3 }
+        val arithmetic = ((intLiteral(5) + 4) * intLiteral(2)) % 3
         assertThat(arithmetic.toString()).isEqualTo("(((5 + 4) * 2) % 3)")
 
-        val caseExpression = with(SqlExpressionBuilder) {
-            case()
-                .When(intLiteral(1) eq 1, stringLiteral("one"))
-                .Else(stringLiteral("other"))
-        }
+        val caseExpression = case()
+            .When(intLiteral(1) eq 1, stringLiteral("one"))
+            .Else(stringLiteral("other"))
         assertThat(caseExpression.toString()).isEqualTo("CASE WHEN 1 = 1 THEN 'one' ELSE 'other' END")
 
-        val coalesced = with(SqlExpressionBuilder) {
-            coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
-        }
+        val coalesced = coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
         assertThat(coalesced.toString()).isEqualTo("COALESCE('primary', 'fallback', 'last')")
 
         val aliased = intLiteral(10).alias("ten")
         assertThat(aliased.toString()).isEqualTo("10 ten")
         assertThat(aliased.aliasOnlyExpression().toString()).isEqualTo("ten")
 
-        val window = with(SqlExpressionBuilder) {
-            rowNumber()
-                .over()
-                .partitionBy(stringLiteral("partition"))
-                .rows(WindowFrameBound.unboundedPreceding())
-        }
+        val window = rowNumber()
+            .over()
+            .partitionBy(stringLiteral("partition"))
+            .rows(WindowFrameBound.unboundedPreceding())
         assertThat(window.toString()).isEqualTo("ROW_NUMBER() OVER(PARTITION BY 'partition' ROWS UNBOUNDED PRECEDING)")
     }
 

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/Exposed_coreTest.kt
@@ -1,0 +1,275 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jetbrains_exposed.exposed_core
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.jetbrains.exposed.v1.core.Alias
+import org.jetbrains.exposed.v1.core.Column
+import org.jetbrains.exposed.v1.core.DatabaseConfig
+import org.jetbrains.exposed.v1.core.ExperimentalKeywordApi
+import org.jetbrains.exposed.v1.core.Index
+import org.jetbrains.exposed.v1.core.Join
+import org.jetbrains.exposed.v1.core.QueryBuilder
+import org.jetbrains.exposed.v1.core.Schema
+import org.jetbrains.exposed.v1.core.SqlExpressionBuilder
+import org.jetbrains.exposed.v1.core.Table
+import org.jetbrains.exposed.v1.core.Version
+import org.jetbrains.exposed.v1.core.WindowFrameBound
+import org.jetbrains.exposed.v1.core.alias
+import org.jetbrains.exposed.v1.core.and
+import org.jetbrains.exposed.v1.core.append
+import org.jetbrains.exposed.v1.core.intLiteral
+import org.jetbrains.exposed.v1.core.intParam
+import org.jetbrains.exposed.v1.core.stringLiteral
+import org.jetbrains.exposed.v1.core.stringParam
+import org.jetbrains.exposed.v1.exceptions.DuplicateColumnException
+import org.junit.jupiter.api.Test
+
+public class ExposedCoreTest {
+    @Test
+    fun tableMetadataTracksColumnsKeysReferencesAndAliases(): Unit {
+        assertThat(Users.tableName).isEqualTo("app.users")
+        assertThat(Users.schemaName).isEqualTo("app")
+        assertThat(Users.columns.map { it.name })
+            .containsExactly("id", "email", "display_name", "status", "score", "active", "generated_token")
+        val primaryKeyColumns: List<Column<*>> = requireNotNull(Users.primaryKey).columns.toList()
+        assertThat(primaryKeyColumns).containsExactly(Users.id)
+        assertThat(Users.autoIncColumn).isEqualTo(Users.id)
+
+        val emailIndex: Index = Users.indices.single()
+        assertThat(emailIndex.customName).isEqualTo("ux_users_email")
+        assertThat(emailIndex.unique).isTrue()
+        assertThat(emailIndex.columns).containsExactly(Users.email)
+
+        assertThat(Memberships.user.referee).isEqualTo(Users.id)
+        assertThat(Memberships.group.referee).isEqualTo(Groups.id)
+        assertThat(Memberships.foreignKeys).hasSize(2)
+
+        val userAlias: Alias<Users> = Users.alias("u")
+        val aliasedEmail: Column<String> = userAlias[Users.email]
+        assertThat(userAlias.tableName).isEqualTo("u")
+        assertThat(userAlias.tableNameWithAlias).isEqualTo("app.users u")
+        assertThat(aliasedEmail.name).isEqualTo(Users.email.name)
+        assertThat(aliasedEmail.table).isEqualTo(userAlias)
+        assertThat(userAlias.originalColumn(aliasedEmail)).isEqualTo(Users.email)
+    }
+
+    @Test
+    fun joinsInferForeignKeysAndExposeCombinedFieldSets(): Unit {
+        val userMembershipJoin: Join = Users.innerJoin(Memberships)
+
+        assertThat(userMembershipJoin.columns).containsExactlyElementsOf(Users.columns + Memberships.columns)
+        assertThat(userMembershipJoin.fields).containsExactlyElementsOf(Users.fields + Memberships.fields)
+        assertThat(userMembershipJoin.alreadyInJoin(Memberships)).isTrue()
+        assertThat(userMembershipJoin.alreadyInJoin(Groups)).isFalse()
+
+        val userMembershipGroupJoin: Join = userMembershipJoin.innerJoin(Groups)
+
+        assertThat(userMembershipGroupJoin.columns)
+            .containsExactlyElementsOf(Users.columns + Memberships.columns + Groups.columns)
+        assertThat(userMembershipGroupJoin.fields)
+            .containsExactlyElementsOf(Users.fields + Memberships.fields + Groups.fields)
+        assertThat(userMembershipGroupJoin.alreadyInJoin(Memberships)).isTrue()
+        assertThat(userMembershipGroupJoin.alreadyInJoin(Groups)).isTrue()
+    }
+
+    @Test
+    fun duplicateColumnNamesAreRejected(): Unit {
+        val table: Table = Table("duplicate_columns")
+        table.integer("id")
+
+        assertThatThrownBy { table.varchar("id", 16) }
+            .isInstanceOf(DuplicateColumnException::class.java)
+            .hasMessageContaining("id")
+            .hasMessageContaining("duplicate_columns")
+    }
+
+    @Test
+    fun columnDefaultsNullabilityAndGeneratedFlagsAreTracked(): Unit {
+        assertThat(Users.email.defaultValueFun == null).isTrue()
+        assertThat(Users.email.defaultValueInDb() == null).isTrue()
+        assertThat(Users.email.columnType.nullable).isFalse()
+
+        assertThat(Users.displayName.columnType.nullable).isTrue()
+        assertThat(Users.status.defaultValueFun?.invoke()).isEqualTo("active")
+        assertThat(Users.status.defaultValueInDb().toString()).isEqualTo("'active'")
+        assertThat(Users.score.defaultValueFun?.invoke()).isEqualTo(7)
+        assertThat(Users.score.defaultValueInDb() == null).isTrue()
+        assertThat(Users.active.defaultValueFun?.invoke()).isTrue()
+        assertThat(Users.active.defaultValueInDb()).isNotNull()
+
+        Users.generatedToken.defaultValueFun?.invoke().let { value: String? ->
+            assertThat(value).isEqualTo("generated")
+        }
+        assertThat(Users.generatedToken.isDatabaseGenerated()).isTrue()
+    }
+
+    @Test
+    fun columnTypesValidateAndFormatValues(): Unit {
+        assertThat(Users.id.columnType.valueFromDB("42")).isEqualTo(42)
+        assertThat(Users.email.columnType.valueFromDB(byteArrayOf(65, 66, 67))).isEqualTo("ABC")
+        assertThat(Users.email.columnType.valueToString("O'Reilly\nGuide")).isEqualTo("'O''Reilly\\nGuide'")
+        assertThat(Users.displayName.columnType.valueToString(null)).isEqualTo("NULL")
+
+        Users.email.columnType.validateValueBeforeUpdate("short@example.org")
+        assertThatThrownBy { Users.email.columnType.validateValueBeforeUpdate("x".repeat(129)) }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("exceeds length")
+
+        assertThatThrownBy { Users.email.columnType.valueToString(null) }
+            .isInstanceOf(IllegalStateException::class.java)
+            .hasMessageContaining("NULL in non-nullable column")
+    }
+
+    @Test
+    fun queryBuilderFormatsInlineSqlAndPreparedArguments(): Unit {
+        val inlineBuilder: QueryBuilder = QueryBuilder(prepared = false)
+        inlineBuilder.append(intLiteral(42), ", ", stringLiteral("O'Reilly"))
+
+        assertThat(inlineBuilder.toString()).isEqualTo("42, 'O''Reilly'")
+        assertThat(inlineBuilder.args).isEmpty()
+
+        val preparedBuilder: QueryBuilder = QueryBuilder(prepared = true)
+        preparedBuilder.append(intParam(7), ", ", stringParam("alpha"))
+
+        assertThat(preparedBuilder.toString()).isEqualTo("?, ?")
+        assertThat(preparedBuilder.args.map { it.second }).containsExactly(7, "alpha")
+    }
+
+    @Test
+    fun expressionDslBuildsComparisonListsAliasesCasesAndWindows(): Unit {
+        val comparison = with(SqlExpressionBuilder) {
+            (intLiteral(1) less intLiteral(3)) and (stringLiteral("kotlin") like "kot%")
+        }
+        assertThat(comparison.toString()).isEqualTo("(1 < 3) AND ('kotlin' LIKE 'kot%')")
+
+        val inList = with(SqlExpressionBuilder) { intLiteral(2) inList listOf(3, 1, 2) }
+        assertThat(inList.toString()).isEqualTo("2 IN (3, 1, 2)")
+
+        val arithmetic = with(SqlExpressionBuilder) { ((intLiteral(5) + 4) * intLiteral(2)) % 3 }
+        assertThat(arithmetic.toString()).isEqualTo("(((5 + 4) * 2) % 3)")
+
+        val caseExpression = with(SqlExpressionBuilder) {
+            case()
+                .When(intLiteral(1) eq 1, stringLiteral("one"))
+                .Else(stringLiteral("other"))
+        }
+        assertThat(caseExpression.toString()).isEqualTo("CASE WHEN 1 = 1 THEN 'one' ELSE 'other' END")
+
+        val coalesced = with(SqlExpressionBuilder) {
+            coalesce(stringLiteral("primary"), stringLiteral("fallback"), stringLiteral("last"))
+        }
+        assertThat(coalesced.toString()).isEqualTo("COALESCE('primary', 'fallback', 'last')")
+
+        val aliased = intLiteral(10).alias("ten")
+        assertThat(aliased.toString()).isEqualTo("10 ten")
+        assertThat(aliased.aliasOnlyExpression().toString()).isEqualTo("ten")
+
+        val window = with(SqlExpressionBuilder) {
+            rowNumber()
+                .over()
+                .partitionBy(stringLiteral("partition"))
+                .rows(WindowFrameBound.unboundedPreceding())
+        }
+        assertThat(window.toString()).isEqualTo("ROW_NUMBER() OVER(PARTITION BY 'partition' ROWS UNBOUNDED PRECEDING)")
+    }
+
+    @Test
+    fun versionParsingComparesSemanticComponents(): Unit {
+        val version: Version = Version.from("1.2.0-beta-4")
+
+        assertThat(version.toString()).isEqualTo("1.2.0")
+        assertThat(version.covers("1.1.9")).isTrue()
+        assertThat(version.covers("1.2.0")).isTrue()
+        assertThat(version.covers("1.2.1")).isFalse()
+        assertThat(Version.from("2").covers(1, 9, 9)).isTrue()
+    }
+
+    @Test
+    @OptIn(ExperimentalKeywordApi::class)
+    fun databaseConfigBuilderAppliesDefaultsCustomizationsAndValidation(): Unit {
+        val defaultConfig: DatabaseConfig = DatabaseConfig()
+
+        assertThat(defaultConfig.useNestedTransactions).isFalse()
+        assertThat(defaultConfig.defaultFetchSize).isNull()
+        assertThat(defaultConfig.defaultIsolationLevel).isEqualTo(-1)
+        assertThat(defaultConfig.defaultMaxAttempts).isEqualTo(3)
+        assertThat(defaultConfig.defaultMinRetryDelay).isZero()
+        assertThat(defaultConfig.defaultMaxRetryDelay).isZero()
+        assertThat(defaultConfig.defaultReadOnly).isFalse()
+        assertThat(defaultConfig.warnLongQueriesDuration).isNull()
+        assertThat(defaultConfig.maxEntitiesToStoreInCachePerEntity).isEqualTo(Int.MAX_VALUE)
+        assertThat(defaultConfig.keepLoadedReferencesOutOfTransaction).isFalse()
+        assertThat(defaultConfig.explicitDialect).isNull()
+        assertThat(defaultConfig.defaultSchema).isNull()
+        assertThat(defaultConfig.logTooMuchResultSetsThreshold).isZero()
+        assertThat(defaultConfig.preserveKeywordCasing).isTrue()
+
+        val schema = Schema("analytics")
+        val customConfig: DatabaseConfig = DatabaseConfig {
+            useNestedTransactions = true
+            defaultFetchSize = 64
+            defaultIsolationLevel = 8
+            defaultMaxAttempts = 2
+            defaultMinRetryDelay = 10L
+            defaultMaxRetryDelay = 20L
+            defaultReadOnly = true
+            warnLongQueriesDuration = 250L
+            maxEntitiesToStoreInCachePerEntity = 5
+            keepLoadedReferencesOutOfTransaction = true
+            defaultSchema = schema
+            logTooMuchResultSetsThreshold = 3
+            preserveKeywordCasing = false
+        }
+
+        assertThat(customConfig.useNestedTransactions).isTrue()
+        assertThat(customConfig.defaultFetchSize).isEqualTo(64)
+        assertThat(customConfig.defaultIsolationLevel).isEqualTo(8)
+        assertThat(customConfig.defaultMaxAttempts).isEqualTo(2)
+        assertThat(customConfig.defaultMinRetryDelay).isEqualTo(10L)
+        assertThat(customConfig.defaultMaxRetryDelay).isEqualTo(20L)
+        assertThat(customConfig.defaultReadOnly).isTrue()
+        assertThat(customConfig.warnLongQueriesDuration).isEqualTo(250L)
+        assertThat(customConfig.maxEntitiesToStoreInCachePerEntity).isEqualTo(5)
+        assertThat(customConfig.keepLoadedReferencesOutOfTransaction).isTrue()
+        assertThat(customConfig.defaultSchema).isEqualTo(schema)
+        assertThat(customConfig.logTooMuchResultSetsThreshold).isEqualTo(3)
+        assertThat(customConfig.preserveKeywordCasing).isFalse()
+
+        assertThatThrownBy { DatabaseConfig { defaultMaxAttempts = 0 } }
+            .isInstanceOf(IllegalArgumentException::class.java)
+            .hasMessageContaining("defaultMaxAttempts")
+    }
+}
+
+private object Users : Table("app.users") {
+    val id = integer("id").autoIncrement()
+    val email = varchar("email", 128).uniqueIndex("ux_users_email")
+    val displayName = varchar("display_name", 64).nullable()
+    val status = varchar("status", 32).default("active")
+    val score = integer("score").clientDefault { 7 }
+    val active = bool("active").default(true)
+    val generatedToken = varchar("generated_token", 32).clientDefault { "generated" }.databaseGenerated()
+
+    override val primaryKey = PrimaryKey(id, name = "pk_users")
+}
+
+private object Groups : Table("groups") {
+    val id = integer("id")
+    val name = varchar("name", 80)
+
+    override val primaryKey = PrimaryKey(id)
+}
+
+private object Memberships : Table("memberships") {
+    val user = reference("user_id", Users.id)
+    val group = reference("group_id", Groups.id)
+    val role = varchar("role", 32).default("member")
+
+    override val primaryKey = PrimaryKey(user, group, name = "pk_memberships")
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/OracleVectorBinderTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/OracleVectorBinderTest.kt
@@ -1,0 +1,119 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jetbrains_exposed.exposed_core
+
+import org.assertj.core.api.Assertions.assertThat
+import org.jetbrains.exposed.v1.core.ArrayColumnType
+import org.jetbrains.exposed.v1.core.DatabaseApi
+import org.jetbrains.exposed.v1.core.DatabaseConfig
+import org.jetbrains.exposed.v1.core.FloatVectorColumnType
+import org.jetbrains.exposed.v1.core.IColumnType
+import org.jetbrains.exposed.v1.core.IntVectorColumnType
+import org.jetbrains.exposed.v1.core.InternalApi
+import org.jetbrains.exposed.v1.core.Transaction
+import org.jetbrains.exposed.v1.core.VectorFormat
+import org.jetbrains.exposed.v1.core.Version
+import org.jetbrains.exposed.v1.core.statements.api.IdentifierManagerApi
+import org.jetbrains.exposed.v1.core.statements.api.PreparedStatementApi
+import org.jetbrains.exposed.v1.core.transactions.TransactionManagerApi
+import org.jetbrains.exposed.v1.core.transactions.withThreadLocalTransaction
+import org.jetbrains.exposed.v1.core.vendors.DatabaseDialect
+import org.jetbrains.exposed.v1.core.vendors.OracleDialect
+import org.junit.jupiter.api.Test
+import java.io.InputStream
+
+public class OracleVectorBinderTest {
+    @Test
+    public fun oracleVectorParametersAreConvertedThroughOracleJdbcVectorFactories(): Unit = withOracleDialect {
+        val statement = CapturingPreparedStatement()
+        val float32ColumnType = FloatVectorColumnType(dimensions = 3, format = VectorFormat.FLOAT32)
+        val float64ColumnType = FloatVectorColumnType(dimensions = 3, format = VectorFormat.FLOAT64)
+        val int8ColumnType = IntVectorColumnType(dimensions = 3, format = VectorFormat.INT8)
+
+        float32ColumnType.setParameter(statement, index = 1, value = floatArrayOf(1.25f, 2.5f, 3.75f))
+        float64ColumnType.setParameter(statement, index = 2, value = doubleArrayOf(1.25, 2.5, 3.75))
+        int8ColumnType.setParameter(statement, index = 3, value = intArrayOf(1, 2, 3))
+
+        assertThat(statement.boundValues.map { it.index }).containsExactly(1, 2, 3)
+        assertThat(statement.boundValues.map { it.value.javaClass.name })
+            .containsExactly("oracle.sql.VECTOR", "oracle.sql.VECTOR", "oracle.sql.VECTOR")
+    }
+}
+
+@OptIn(InternalApi::class)
+private fun withOracleDialect(block: () -> Unit): Unit {
+    withThreadLocalTransaction(FakeTransaction(FakeDatabase(OracleDialect())), block)
+}
+
+private data class BoundValue(
+    val index: Int,
+    val value: Any,
+)
+
+private class CapturingPreparedStatement : PreparedStatementApi {
+    val boundValues = mutableListOf<BoundValue>()
+
+    override fun set(index: Int, value: Any, columnType: IColumnType<*>): Unit {
+        boundValues += BoundValue(index, value)
+    }
+
+    override fun setNull(index: Int, columnType: IColumnType<*>): Unit {
+        throw AssertionError("Oracle vector parameters must be converted to non-null VECTOR instances")
+    }
+
+    override fun setInputStream(index: Int, inputStream: InputStream, setAsBlobObject: Boolean): Unit {
+        throw AssertionError("Vector columns must not bind input streams")
+    }
+
+    override fun setArray(index: Int, type: ArrayColumnType<*, *>, array: Array<*>): Unit {
+        throw AssertionError("Vector columns must not bind SQL arrays")
+    }
+}
+
+private class FakeTransaction(override val db: DatabaseApi) : Transaction() {
+    override val transactionManager: TransactionManagerApi = FakeTransactionManager()
+    override val readOnly: Boolean = false
+    override val outerTransaction: Transaction? = null
+}
+
+private class FakeTransactionManager : TransactionManagerApi {
+    override var defaultReadOnly: Boolean = false
+    override var defaultMaxAttempts: Int = 1
+    override var defaultMinRetryDelay: Long = 0L
+    override var defaultMaxRetryDelay: Long = 0L
+}
+
+private class FakeDatabase(override val dialect: DatabaseDialect) : DatabaseApi(
+    resolvedVendor = "oracle",
+    config = DatabaseConfig()
+) {
+    override val url: String = "jdbc:oracle:thin:@localhost:1521/test"
+    override val vendor: String = "oracle"
+    override val dialectMode: String? = null
+    override val version: Version = Version.from("23.0.0")
+    override val fullVersion: String = "23.0.0"
+    override val supportsAlterTableWithAddColumn: Boolean = true
+    override val supportsAlterTableWithDropColumn: Boolean = true
+    override val supportsMultipleResultSets: Boolean = true
+    override val supportsSelectForUpdate: Boolean = true
+    override val identifierManager: IdentifierManagerApi = FakeIdentifierManager()
+}
+
+private class FakeIdentifierManager : IdentifierManagerApi() {
+    override val quoteString: String = "\""
+    override val isUpperCaseIdentifiers: Boolean = false
+    override val isUpperCaseQuotedIdentifiers: Boolean = false
+    override val isLowerCaseIdentifiers: Boolean = false
+    override val isLowerCaseQuotedIdentifiers: Boolean = false
+    override val supportsMixedIdentifiers: Boolean = true
+    override val supportsMixedQuotedIdentifiers: Boolean = true
+    override val extraNameCharacters: String = ""
+    override val oracleVersion: OracleVersion = OracleVersion.NonOracle
+    override val maxColumnNameLength: Int = 128
+
+    override fun dbKeywords(): List<String> = emptyList()
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/SqlServerVectorBinderTest.kt
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/src/test/kotlin/org_jetbrains_exposed/exposed_core/SqlServerVectorBinderTest.kt
@@ -1,0 +1,138 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jetbrains_exposed.exposed_core
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.jetbrains.exposed.v1.core.ArrayColumnType
+import org.jetbrains.exposed.v1.core.DatabaseApi
+import org.jetbrains.exposed.v1.core.DatabaseConfig
+import org.jetbrains.exposed.v1.core.FloatVectorColumnType
+import org.jetbrains.exposed.v1.core.IColumnType
+import org.jetbrains.exposed.v1.core.IntVectorColumnType
+import org.jetbrains.exposed.v1.core.InternalApi
+import org.jetbrains.exposed.v1.core.Transaction
+import org.jetbrains.exposed.v1.core.VectorFormat
+import org.jetbrains.exposed.v1.core.Version
+import org.jetbrains.exposed.v1.core.statements.api.IdentifierManagerApi
+import org.jetbrains.exposed.v1.core.statements.api.PreparedStatementApi
+import org.jetbrains.exposed.v1.core.statements.api.RowApi
+import org.jetbrains.exposed.v1.core.transactions.TransactionManagerApi
+import org.jetbrains.exposed.v1.core.transactions.withThreadLocalTransaction
+import org.jetbrains.exposed.v1.core.vendors.DatabaseDialect
+import org.jetbrains.exposed.v1.core.vendors.SQLServerDialect
+import org.junit.jupiter.api.Test
+import java.io.InputStream
+
+public class SqlServerVectorBinderTest {
+    @Test
+    public fun sqlServerVectorParametersAreConvertedThroughJdbcVectorConstructorAndReadBack(): Unit =
+        withSqlServerDialect {
+            val statement = SqlServerCapturingPreparedStatement()
+            val floatColumnType = FloatVectorColumnType(dimensions = 3, format = VectorFormat.FLOAT32)
+            val intColumnType = IntVectorColumnType(dimensions = 3, format = VectorFormat.INT8)
+
+            floatColumnType.setParameter(statement, index = 1, value = floatArrayOf(1.25f, 2.5f, 3.75f))
+            assertThatThrownBy { intColumnType.setParameter(statement, index = 2, value = intArrayOf(1, 2, 3)) }
+                .hasRootCauseInstanceOf(IllegalArgumentException::class.java)
+
+            assertThat(statement.boundValues.map { it.index }).containsExactly(1)
+            assertThat(statement.boundValues.map { it.value.javaClass.name }).containsExactly("microsoft.sql.Vector")
+
+            val floatData = floatColumnType.readObject(SqlServerVectorRow(statement.boundValues[0].value), index = 1)
+
+            assertThat(floatData as Array<*>).containsExactly(1.25f, 2.5f, 3.75f)
+        }
+}
+
+@OptIn(InternalApi::class)
+private fun withSqlServerDialect(block: () -> Unit): Unit {
+    withThreadLocalTransaction(SqlServerFakeTransaction(SqlServerFakeDatabase(SQLServerDialect())), block)
+}
+
+private data class SqlServerBoundValue(
+    val index: Int,
+    val value: Any,
+)
+
+private class SqlServerCapturingPreparedStatement : PreparedStatementApi {
+    val boundValues = mutableListOf<SqlServerBoundValue>()
+
+    override fun set(index: Int, value: Any, columnType: IColumnType<*>): Unit {
+        boundValues += SqlServerBoundValue(index, value)
+    }
+
+    override fun setNull(index: Int, columnType: IColumnType<*>): Unit {
+        throw AssertionError("SQL Server vector parameters must be converted to non-null Vector instances")
+    }
+
+    override fun setInputStream(index: Int, inputStream: InputStream, setAsBlobObject: Boolean): Unit {
+        throw AssertionError("Vector columns must not bind input streams")
+    }
+
+    override fun setArray(index: Int, type: ArrayColumnType<*, *>, array: Array<*>): Unit {
+        throw AssertionError("Vector columns must not bind SQL arrays")
+    }
+}
+
+private class SqlServerVectorRow(private val vector: Any) : RowApi {
+    override fun getObject(index: Int): Any? = vector
+
+    override fun getObject(name: String): Any? = vector
+
+    override fun <T> getObject(index: Int, type: Class<T>): T? =
+        if (type.isInstance(vector)) type.cast(vector) else null
+
+    override fun <T> getObject(name: String, type: Class<T>): T? =
+        if (type.isInstance(vector)) type.cast(vector) else null
+
+    override fun getString(index: Int): String? = vector.toString()
+}
+
+private class SqlServerFakeTransaction(override val db: DatabaseApi) : Transaction() {
+    override val transactionManager: TransactionManagerApi = SqlServerFakeTransactionManager()
+    override val readOnly: Boolean = false
+    override val outerTransaction: Transaction? = null
+}
+
+private class SqlServerFakeTransactionManager : TransactionManagerApi {
+    override var defaultReadOnly: Boolean = false
+    override var defaultMaxAttempts: Int = 1
+    override var defaultMinRetryDelay: Long = 0L
+    override var defaultMaxRetryDelay: Long = 0L
+}
+
+private class SqlServerFakeDatabase(override val dialect: DatabaseDialect) : DatabaseApi(
+    resolvedVendor = "sqlserver",
+    config = DatabaseConfig()
+) {
+    override val url: String = "jdbc:sqlserver://localhost;databaseName=test"
+    override val vendor: String = "sqlserver"
+    override val dialectMode: String? = null
+    override val version: Version = Version.from("17.0.0")
+    override val fullVersion: String = "17.0.0"
+    override val supportsAlterTableWithAddColumn: Boolean = true
+    override val supportsAlterTableWithDropColumn: Boolean = true
+    override val supportsMultipleResultSets: Boolean = true
+    override val supportsSelectForUpdate: Boolean = true
+    override val identifierManager: IdentifierManagerApi = SqlServerFakeIdentifierManager()
+}
+
+private class SqlServerFakeIdentifierManager : IdentifierManagerApi() {
+    override val quoteString: String = "\""
+    override val isUpperCaseIdentifiers: Boolean = false
+    override val isUpperCaseQuotedIdentifiers: Boolean = false
+    override val isLowerCaseIdentifiers: Boolean = false
+    override val isLowerCaseQuotedIdentifiers: Boolean = false
+    override val supportsMixedIdentifiers: Boolean = true
+    override val supportsMixedQuotedIdentifiers: Boolean = true
+    override val extraNameCharacters: String = ""
+    override val oracleVersion: OracleVersion = OracleVersion.NonOracle
+    override val maxColumnNameLength: Int = 128
+
+    override fun dbKeywords(): List<String> = emptyList()
+}

--- a/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/user-code-filter.json
+++ b/tests/src/org.jetbrains.exposed/exposed-core/1.3.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.jetbrains.exposed.v1.**"
+    },
+    {
+      "includeClasses" : "org_jetbrains_exposed.exposed_core.**"
+    }
+  ]
+}

